### DISCUSSION TITLE
about prompt template

### DISCUSSION THREAD

**Discussion by lileltp** [FlowiseAI/Flowise#1435] [Q&A]
I have defined multiple variable values in the prompt template, for example: 'this is {profile} \n\n this is {topic}', but now I can only choose {{question}} as a single input in the API. This results in the values of profile and topic being the same. How can I pass two separate values in the API to differentiate between profile and topic?

**Comment by Jaredude** [comment]
Hope this helps...

If you want the prompt to return "this is {profile} \n\n this is {topic}", the LLM probably won't do that for you even with a temperature of 1.

There are two ways you can pass multiple variables into an API endpoint:

1. Include both values in the question: e.g. format your question as follows "profile: Scott, topic: cats"
2. Use x-www-form-urlencoded instead of application/json and pass the multiple variables as url encoded values of "question": e.g. question=profile%3DScott%26topic%3Dcats

Then you'll need the Format Prompt Values for the Prompt Template to have 2 items: profile and topic.

Using a prompt template along the following will show that you can get both values from "question": Write a brief sentence about {topic} that includes {profile}

If you want to digest those values in another chain, you should then be able to use a Structured Output Parser to then get a json payload that includes the "answer" (required) and the original profile and topic values.

Here's a Postman example output:
<img width="1014" alt="Screenshot 2023-12-29 at 3 07 20 PM" src="https://github.com/FlowiseAI/Flowise/assets/10228950/c4e90290-5a90-4e5e-ad87-da727d759da0">

Here's the prompt template using {{question}} for both profile and topic and the language model is able to determine which should be assigned to which:
<img width="643" alt="Screenshot 2023-12-29 at 3 15 40 PM" src="https://github.com/FlowiseAI/Flowise/assets/10228950/c47beb2c-361d-4829-a054-bad9ee700042">

**Reply by lileltp** [reply]
![image](https://github.com/FlowiseAI/Flowise/assets/4696657/f7b39c7a-0e21-4458-bcd9-89475f076a73)

**Comment by lileltp** [comment]
{
	"question": "用500字",
	"overrideConfig":{
		"systemMessagePrompt":"编写一个故事关于Scott家的cat"
	}
}
I use this get it
